Makefile

```diff
@@ -111,23 +119,37 @@ generate: vendor controller-gen ## Generates types deepcopy code
 manifests: controller-gen kustomize ## Generates the manifests in $PROJECT_DIR/install
 	controller-gen crd:crdVersions=v1 rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=install/crd output:rbac:artifacts:config=install/rbac && $(KUSTOMIZE) build install > $(AUTHORINO_MANIFESTS)
 	$(MAKE) patch-webhook
-
+	
+run:GIT_SHA=$(shell git rev-parse HEAD)
+run:DIRTY=$(shell $(PROJECT_DIR)/hack/check-git-dirty.sh || echo "unknown")
+run:$(shell $(PROJECT_DIR)/hack/check-version.sh)
+run:VERSION=$(shell $(YQ) '.build.version' build.yaml)
 run: generate manifests ## Runs the application against the Kubernetes cluster configured in ~/.kube/config
-	go run -ldflags "-X main.version=$(VERSION)" ./main.go server
+	go run -ldflags "-X main.version=$(VERSION) -X main.gitSHA=${GIT_SHA} -X main.dirty=${DIRTY}" ./main.go server
 
+build:GIT_SHA=$(shell git rev-parse HEAD)
+build:DIRTY=$(shell $(PROJECT_DIR)/hack/check-git-dirty.sh || echo "unknown")
+build: $(shell $(PROJECT_DIR)/hack/check-version.sh)
+build:VERSION=$(shell $(YQ) '.build.version' build.yaml)
 build: generate ## Builds the manager binary
-	CGO_ENABLED=0 GO111MODULE=on go build -a -ldflags "-X main.version=$(VERSION)" -o bin/authorino main.go
+	CGO_ENABLED=0 GO111MODULE=on go build -a -ldflags "-X main.version=$(VERSION) -X main.gitSHA=${GIT_SHA} -X main.dirty=${DIRTY}" -o bin/authorino main.go
 
 IMAGE_REPO ?= authorino
+VERSION ?= $(GIT_SHA)
 using_semantic_version := $(shell [[ $(VERSION) =~ ^[0-9]+\.[0-9]+\.[0-9]+(-.+)?$$ ]] && echo "true")
```

Comment:
It looks like `using_semantic_version` will never be true here when VERSION is defined equal to `$(GIT_SHA)`. What use case does this cover?

Reply:
We can remove that line since it doesn't cover the use case of semantic versioning